### ISSUE TITLE
null mixer broken

### ISSUE THREAD

**Issue by quantenschaum** [MusicPlayerDaemon/MPD#138]
The [doc](https://www.musicpd.org/doc/user/config_audio_outputs.html) says

 > the "null" mixer (null; allows setting the volume, but with no effect; this can be used as a trick to implement an external mixer)

So I would expect, that I can set and get the volume from MPD, but actual playback is always at volume=100.

When I set `mixer_type "null"` in `mpd.conf` then I get

```
$ mpc volume 
volume: n/a
$ mpc volume 50
mpd error: problems setting volume
```

This is what `mixer_type "none"` should do, but not `null`. With this behaviour implementing an [external mixer](https://www.musicpd.org/doc/user/client_hacks.html#external_mixer) is impossible.

I have this issue on a Raspberry Pi running Raspbian Stretch and Music Player Daemon 0.20.10

**Comment by mistepien** [comment]
root@shadow:/tmp/root# mpc.bin volume
volume: 74%
root@shadow:/tmp/root# mpc.bin volume 100
volume:100%   repeat: off   random: off   single: off   consume: off
root@shadow:/tmp/root# mpc volume
volume:100%
root@shadow:/tmp/root# mpc volume 5
volume:  5%   repeat: off   random: off   single: off   consume: off
root@shadow:/tmp/root# mpc volume
volume:  5%

null mixer is set. it works like that since it was added to mpd

**Comment by quantenschaum** [comment]
This is the expected behaviour, yes. But in my case it does not work. What is (am I doing) wrong?

mpd.conf
```
zeroconf_name           "raspberrypi"
bind_to_address         "any"
bind_to_address         "/run/mpd/socket"
music_directory         "/media"
playlist_directory      "/media/.mpd/playlists"
db_file                 "/media/.mpd/tag_cache"
state_file              "/media/.mpd/state"
sticker_file            "/media/.mpd/sticker.sql"
#log_file                "syslog"
log_file                "/dev/null"

filesystem_charset      "UTF-8"
id3v1_encoding          "UTF-8"

samplerate_converter    "2"
replaygain              "auto"

audio_output {
 type "alsa"
 name "alsa"
# mixer_type "software"
  mixer_type "null"
# mixer_control "Master"
# format "48000:24:2"
 dop "no"
}
```

and 
```
root@raspberrypi:~ # mpc
volume: n/a   repeat: off   random: off   single: off   consume: off
root@raspberrypi:~ # mpc vol 66
mpd error: problems setting volume
root@raspberrypi:~ # mpc vol
volume: n/a
```

**Comment by quantenschaum** [comment]
@mistepien What is an "alsa section"?

**Comment by MaxKellermann** [comment]
Paste the output of `mpc outputs`

**Comment by quantenschaum** [comment]
```
root@raspberrypi:~ # mpc outputs
Output 1 (alsa) is enabled
```

**Comment by quantenschaum** [comment]
with a minimal configuartion

```
audio_output {
 type "null"
 name "foo"
 mixer_type "null"
}
```

```
root@raspberrypi:~ # mpc outputs; mpc
Output 1 (foo) is enabled
volume: n/a   repeat: off   random: off   single: off   consume: off
```

**Comment by quantenschaum** [comment]
@mistepien Could you please post your `mpd.conf`.

**Comment by mistepien** [comment]
ok, my mpd.conf is generated by script (autodetection of installed audio cards)

[mpd.conf.txt](https://github.com/MusicPlayerDaemon/MPD/files/1466906/mpd.conf.txt)

**Comment by MaxKellermann** [comment]
No matter what I do, I fail to repdroduce it, and the MPD source code looks like what you pasted is impossible. To check one remaining theory, paste the output of:

    $(readlink /proc/$(pidof mpd)/exe) --version

**Comment by MaxKellermann** [comment]
And:

    mpc version

**Comment by quantenschaum** [comment]
```
root@raspberrypi:~ # mpc version
mpd version: 0.19.0
```

```
root@raspberrypi:~ # $(readlink /proc/$(pidof mpd)/exe) --version
Music Player Daemon 0.19.21
...
```

Ahh! There it is! `make install` puts it to `/usr/local/bin/mpd`, the one from the distro still is in `/usr/bin/mpd`. So it's working in 0.20.11, but is broken in 0.19.21.

Thanks!

**Comment by MaxKellermann** [comment]
Not broken - just not yet implemented. The feature was implemented in version 0.20.
